File: SampleDataBinding/app/src/main/java/net/kokohadoko/instil/sampledatabinding/MainActivity.java
package net.kokohadoko.instil.sampledatabinding;

import android.databinding.DataBindingUtil;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import net.kokohadoko.instil.sampledatabinding.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);

        User user = new User("Test", "User");
        binding.setUser(user);

    }
}

File: SampleDataBindingKotlin/app/src/main/java/net/kokohadoko/instil/sampledatabindingkotlin/MainActivity.kt
package net.kokohadoko.instil.sampledatabindingkotlin

import android.databinding.DataBindingUtil
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import net.kokohadoko.instil.sampledatabindingkotlin.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main)
        val user = User("Test", "User")
        binding.user = user
    }
}
